starport/services/network/blockchain.go

```diff
@@ -94,24 +104,89 @@ func (b *Blockchain) Init(ctx context.Context) error {
 	}
 
 	// cleanup home dir of app if exists.
-	if err := xos.RemoveAllUnderHome(chainHome); err != nil {
+	if err := os.RemoveAll(chainHome); err != nil {
 		return err
 	}
 
+	// build the chain and initialize it with a new validator key
+	b.builder.ev.Send(events.New(events.StatusOngoing, "Building the blockchain"))
 	if _, err := b.chain.Build(ctx, ""); err != nil {
 		return err
 	}
-
+	b.builder.ev.Send(events.New(events.StatusDone, "Blockchain built"))
+	b.builder.ev.Send(events.New(events.StatusOngoing, "Initializing the blockchain"))
 	if err := b.chain.Init(ctx, false); err != nil {
 		return err
 	}
-
 	b.builder.ev.Send(events.New(events.StatusDone, "Blockchain initialized"))
+
+	// write the custom genesis if a genesis URL is provided
+	if b.genesisURL != "" {
+		genesis, hash, err := genesisAndHashFromURL(ctx, b.genesisURL)
+		if err != nil {
+			return err
+		}
+		if hash != b.genesisHash {
+			return fmt.Errorf("genesis from URL %s is invalid. Expected hash %s, actual hash %s", b.genesisURL, b.genesisHash, hash)
+		}
+		genesisPath, err := b.chain.GenesisPath()
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(genesisPath, genesis, 0644); err != nil {
+			return err
+		}
+	}
+
 	b.isInitialized = true
 
 	return nil
 }
 
+// InitAccount initializes an account for the blockchain and issue a gentx in config/gentx/gentx.json
+func (b *Blockchain) InitAccount(
+	ctx context.Context,
+	v chain.Validator,
+	keyName,
+	mnemonic string,
+) (chaincmdrunner.Account, string, error) {
+	if !b.isInitialized {
+		return chaincmdrunner.Account{}, "", errors.New("the blockchain must be initialized to initialize an account")
```

Comment:
Or we can just run blockchain.Init?

Reply:
We can. However, we would ask validator info and then build the chain which is less good in a UX point of view IMO